src/Database/Statement/StatementDecorator.php

```diff
@@ -35,6 +35,25 @@ class StatementDecorator implements StatementInterface, Countable, IteratorAggre
 
     use TypeConverterTrait;
 
+    /**
+     * Used to designate that numeric indexes be returned in a result when calling fetch methods
+     *
+     * @var string
+     */
+    const FETCH_TYPE_NUM = 'num';
+    /**
+     * Used to designate that associated array be returned in a result when calling fetch methods
+     *
+     * @var string
+     */
+    const FETCH_TYPE_ASSOC = 'assoc';
+    /**
+     * Used to designate that objects of \stdClass be returned in a result when calling fetch methods
+     *
+     * @var string
+     */
+    const FETCH_TYPE_OBJ = 'obj';
```

Comment:
do you use this mode in your projects?

I think having it creates a lot of confusion, and makes certain things like using collection methods more difficult.

Reply:
Should we map results into stdclass like object that implements arrayaccess as well?

Reply:
I made the assumption that if you wanted to call one of the methods you probably were wanting lower level pdo access. This pull request doesn't fully address it but does provide a couple options and abstracts the the strings (num, assoc) out of the way.

Reply:
I like the other changes, but I’m unsure about this one.